packages/engine-server/src/workspace/service.ts

```diff
@@ -879,6 +902,11 @@ export class WorkspaceService implements Disposable, IWorkspaceService {
               message: `error pulling vault${stderr}`,
               payload: { err, repoPath: repo },
             });
+          } finally {
+            // Try to restore changes if we stashed them, even if there were errors. We don't want to lose the users changes.
+            if (stashed) {
+              git.stashApplyCommit(stashed);
```

Comment:
can this fail?

Reply:
Tested (and added automated tests) if there are conflicting changes. It doesn't fail, it just creates a merge conflict (`<<<< updated upstream...`). I'll add some code to detect that, and warn the user that a merge conflict happened.

I'm not sure if there are other cases where it could fail, let me know if you have something in mind and I'll add a test for that too.

Reply:
for merge conflict, can we actually fail and warn the user? would be better to be explicit then to commit a conflict